app/src/main/java/org/mifos/mobile/ui/fragments/AddGuarantorFragment.kt

```diff
@@ -39,20 +40,27 @@ class AddGuarantorFragment : BaseFragment() {
 
     private val viewModel: AddGuarantorViewModel by viewModels()
 
-    var guarantorTypeAdapter: ArrayAdapter<String?>? = null
+    private var guarantorTypeAdapter: ArrayAdapter<String?>? = null
     var template: GuarantorTemplatePayload? = null
     var payload: GuarantorPayload? = null
     private var guarantorState: GuarantorState? = null
     private var guarantorApplicationPayload: GuarantorApplicationPayload? = null
     var loanId: Long? = 0
-    var index: Int? = 0
+    private var index: Int? = 0
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         if (arguments != null) {
             loanId = arguments?.getLong(Constants.LOAN_ID)
-            guarantorState = requireArguments()
-                .getSerializable(Constants.GUARANTOR_STATE) as GuarantorState
-            payload = arguments?.getParcelable(Constants.PAYLOAD)
+            guarantorState = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
```

Comment:
is it necessary to check the Build version ?

Reply:
> is it necessary to check the Build version ?

Yes because the min SDK version is 22 and the above code is deprecated for newer Android api versions so I added a check.

Reply:
> > is it necessary to check the Build version ?
> 
> Yes because the min SDK version is 22 and the above code is deprecated for newer Android api versions i.e. Android Tiramisu onwards, so I added a check.

Reply:
@PratyushSingh07 I have added the changes of formatting the code.